File: interviewprep/lib/core/models/activity_history.dart
class ActivityHistory {
  final String id;
  final String utilisateurId;
  final String type;
  final String message;
  final Map<String, dynamic>? metadata;
  final DateTime creeLe;
  final DateTime modifieLe;

  const ActivityHistory({
    required this.id,
    required this.utilisateurId,
    required this.type,
    required this.message,
    required this.metadata,
    required this.creeLe,
    required this.modifieLe,
  });

  factory ActivityHistory.fromJson(Map<String, dynamic> json) {
    return ActivityHistory(
      id: json['id'] as String,
      utilisateurId: json['utilisateur_id'] as String,
      type: json['type'] as String,
      message: json['message'] as String,
      metadata: (json['metadata'] as Map?)?.cast<String, dynamic>(),
      creeLe: DateTime.parse(json['cree_le'] as String),
      modifieLe: DateTime.parse(json['modifie_le'] as String),
    );
  }
}

File: backend/app/models/activity_history.py
"""Modèle SQLAlchemy pour l'historique des activités utilisateur."""

from sqlalchemy import Column, ForeignKey, Index, JSON, String, Uuid
from sqlalchemy.orm import relationship

from app.models.base import BaseModel


class ActivityHistory(BaseModel):
    """Entrée d'historique d'activité associée à un utilisateur."""

    __tablename__ = "activity_histories"

    utilisateur_id = Column(
        Uuid(as_uuid=True),
        ForeignKey("users.id"),
        nullable=False,
        index=True,
    )
    type = Column("type", String(80), nullable=False, index=True)
    message = Column(String(500), nullable=False)
    metadata_ = Column("metadata", JSON, nullable=True, default=dict)

    utilisateur = relationship("User", back_populates="activity_histories")

    __table_args__ = (
        Index("idx_activity_histories_user_created", "utilisateur_id", "cree_le"),
        Index("idx_activity_histories_type", "type"),
    )
